cmd/kube-proxy/app/server.go

```diff
@@ -572,13 +572,16 @@ func (s *ProxyServer) Run(ctx context.Context) error {
 		return err
 	}
 
-	labelSelector := labels.NewSelector()
-	labelSelector = labelSelector.Add(*noProxyName, *noHeadlessEndpoints)
+	labelSelectorNoProxyName := labels.NewSelector().Add(*noProxyName)
+	labelSelectorNoHeadlessEndpoints := labels.NewSelector().Add(*noHeadlessEndpoints)
 
-	// Make informers that filter out objects that want a non-default service proxy.
-	informerFactory := informers.NewSharedInformerFactoryWithOptions(s.Client, s.Config.ConfigSyncPeriod.Duration,
+	// Make informer that contains no filters
+	informerFactory := informers.NewSharedInformerFactoryWithOptions(s.Client, s.Config.ConfigSyncPeriod.Duration)
+
+	// Make informers that filter out objects that do not contain a service.kubernetes.io/headless label
+	endPointSlicesInformerFactory := informers.NewSharedInformerFactoryWithOptions(s.Client, s.Config.ConfigSyncPeriod.Duration,
```

Comment:
oh, actually, since this isn't going to merge until post-1.35 anyway, you could fix this name: we never capitalize the "p" in "endpoint".
(and also, I guess, it should be "endpointSliceInfomerFactory", not "endpointSlice**s**InformerFactory" (compare "serviceInformerFactory"))